jni/events/sampleevent.cpp

```diff
@@ -182,17 +176,197 @@ void SampleEvent::setSample( AudioBuffer* sampleBuffer )
         _locked = false;
 }
 
-void SampleEvent::mixBuffer( AudioBuffer* outputBuffer, int bufferPos, int minBufferPosition, int maxBufferPosition,
+float SampleEvent::getPlaybackRate()
+{
+    return _playbackRate;
+}
+
+void SampleEvent::setPlaybackRate( float value )
+{
+    // allow only 100x slowdown and speed up
+    _playbackRate = std::max( 0.01f, std::min( 100.f, value ));
+}
+
+int SampleEvent::getEventLength()
+{
+    return ( _playbackRate == 1.f ) ? _eventLength : ( int )( _eventLength / _playbackRate );
+}
+
```

Comment:
Event length is transformed when a custom playback rate is used. Update the getter so we don't have to change the Sequencers interface for requesting events for playback.